parser/src/ast/boundary_constraints.rs

```diff
@@ -8,8 +8,7 @@ use std::fmt::Display;
 
 #[derive(Debug, Eq, PartialEq)]
 pub enum BoundaryStmt {
-    Constraint(BoundaryConstraint),
-    ConstraintComprehension(BoundaryConstraint, ComprehensionContext),
+    Constraint(BoundaryConstraint, Option<ComprehensionContext>),
```

Comment:
I think maybe we should put the comprehension context into the boundary constraint, like this:

```
#[derive(Debug, Eq, PartialEq)]
pub enum BoundaryStmt {
    Constraint(BoundaryConstraint),
    VariableBinding(VariableBinding),
}

/// Stores the expression corresponding to the boundary constraint.
#[derive(Debug, Eq, PartialEq)]
pub struct BoundaryConstraint {
    access: SymbolAccess,
    boundary: Boundary,
    value: Expression,
    comprehension_context: Option<ComprehensionContext>,
}
```